pre_commit/languages/docker.py

```diff
@@ -34,6 +38,9 @@ def _get_container_id() -> str:
         for line in f.readlines():
             if line.split(b':')[1] == b'cpuset':
                 return os.path.basename(line.split(b':')[2]).strip().decode()
+    hostname = socket.gethostname()
+    if re.match(r'^[0-9a-f]{12}$', hostname):
```

Comment:
this isn't reliable.  hostname can be set arbitrarily both inside and outside of docker 

a similar approach has already been rejected

Reply:
yes, that why I tried to at least ensure it looks like some container ID.

if not in a container, then this code won't be executed, so not problem here.

if in a container, at best, the hostname is the container ID and it works which is an improvement.

at worst, the hostname is not a container ID, then it does not work, which is not worse than the actual behavior.

Reply:
in case it's not clear I'm not going to accept this code for the same reasons I've rejected it in the past

Reply:
that was not clear indeed !